charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -74,16 +74,17 @@ spec:
             {{- if .Values.manager.featureGates }}
             - --feature-gates={{ .Values.manager.featureGates }}
             {{- end }}
-            {{- if .Values.manager.createRbacPermissions }}
-            - --create-rbac-permissions
-            {{- end }}
             {{-  if .Values.manager.extraArgs  }}
             {{- .Values.manager.extraArgs | toYaml | nindent 12 }}
             {{-  end  }}
           command:
             - /manager
-          {{- if .Values.manager.env }}
           env:
+            - name: SERVICE_ACCOUNT_NAME
```

Comment:
instead of `manager.createRbacPermissions` setting the RBAC flag, should we only add this env var when `manager.createRbacPermissions: true`?

Reply:
Put it under condition, but it works because of `env` is required based on the schema, otherwise we need to redesign this part with env variables.